src/routes/admin.ts

```diff
@@ -0,0 +1,74 @@
+import express, { NextFunction } from "express";
+import Joi from "joi";
+
+import * as AdminService from "../services/admin";
+
+export const adminRouter = express.Router();
+
+adminRouter.post(
+  "/login",
+  async function (
+    req: express.Request,
+    res: express.Response,
+    next: NextFunction
+  ) {
+    try {
+      const adminExis = await AdminService.checkAdmin(req.body.email);
+
+      if (!adminExis) {
+        throw new Error("Usual user will not be allowed to log in");
+      }
+      next();
+    } catch (error) {
+      return res.status(500).json({ error: error?.message || "Server error" });
+    }
+  },
```

Comment:
tách ra thành 1 cái middleware riêng. r dùng lại cho toàn bộ route admin. 
hiện tại thì đúng nhưng chỉ dùng được 1 route ở login thôi.